index.bs

```diff
@@ -16,13 +16,439 @@ WPT Display: closed
+        1.  [=header list/Set a structured field value=] given ("`Sec-Fetch-Storage-Access`", |value|) in |request|'s [=request/header list=].
+    </ol>
+</div>
+
+
+The `Activate-Storage-Access` HTTP Response Header {#activate-storage-access-header}
+-------------------------------------------------------------------------------------
+
+The <dfn http-header export>`Activate-Storage-Access`</dfn> HTTP response header
+allows a server to opt in to accessing its cookies in a cross-site request
+context. It is a [=Structured Field=] [=structured field/item=] whose value MUST be a [=structured
+field/token=]. [[!I-D.structured-field-values-for-http]] Its ABNF is:
+
+```
+Activate-Storage-Access = sf-item
+```
+
+Valid `Activate-Storage-Access` values include "`load`" and "`retry`". In order
+to support forward-compatibility with as-yet-unknown semantics, servers SHOULD
+ignore this header if it contains an invalid value.
+
+The following parameter is defined:
+
+* A parameter whose key is "`allowed-origin`", and whose value is a [=structured field/string=]. See below for processing requirements.
+
+<pre class="example" id="activate-storage-access-usage">
+
+// The server's response requests that the user agent activate storage access
+// before continuing with the load of the resource.
+Activate-Storage-Access: load
+
+// The server's response requests that the user agent activate storage access,
+// then retry the request. The "allowed-origin" parameter allowlists the
+// request's origin.
+Activate-Storage-Access: retry; allowed-origin="https://foo.bar"
+
+// Same as above, but using a wildcard instead of explicitly naming the request's Origin.
+Activate-Storage-Access: retry; allowed-origin=*
+</pre>
+
+<div algorithm>
+    To <dfn export>perform a storage access retry check</dfn> for a [=request=] |request| and [=response=] |response|, run the following steps:
+
+    <ol class="algorithm">
+        1. If |request|'s [=request/credentials mode=] [=string/is=] not "`include`", return failure.
+        1. If |request|'s [=request/eligible for storage-access=] is true, return failure.
+        1. Let |storageAccessStatus| be |request|'s [=request/storage access status=].
+        1. If |storageAccessStatus| [=string/is=] not "<code>[=storage access status/inactive=]</code>", return failure.
+        1. Let |parsedHeader| be the result of [=header list/get a structured field value|getting a structured field value=] given "`Activate-Storage-Access`" and "`item`" from |response|'s [=response/header list=].
+        1. If |parsedHeader| is null, return failure.
+        1. Let (|value|, |params|) be |parsedHeader|.
+        1. If |value| is not a Token, return failure.
+        1. If |value| [=string/is=] not "`retry`", return failure.
+        1. If |params|["allowed-origin"] does not exist, return failure.
+        1. Let |allowedOrigin| be |params|["allowed-origin"].
+        1. If |allowedOrigin| is a Token whose value is "`*`", return success.
```

Comment:
I'm not a structured headers expert but as I understand it tokens are separate types than strings so maybe just move the "is not a string" check below up one line and remove the "is a token" here?

https://httpwg.org/specs/rfc9651.html#token